File: app/src/main/java/com/lambdaschool/congressdata/ProfileRepository.java
package com.lambdaschool.congressdata;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.util.Log;

import com.lambdaschool.congressdataapiaccess.CongressDao;

public class ProfileRepository {
    public static LiveData<CongresspersonProfile> getProfile(final String id) {
        Log.i("Repository", "Retreiving Data");
        final MutableLiveData<CongresspersonProfile> profileLiveData = new MutableLiveData<>();

        CongresspersonProfile profile = new CongresspersonProfile(CongressDao.getMemberDetails(id));
        profile.setImage(CongressDao.getImage(profile.getId()));
        profileLiveData.setValue(profile);

        return profileLiveData;
    }
}

File: app/src/main/java/com/lambdaschool/congressdata/ProfileRepository.kt
package com.lambdaschool.congressdata

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.util.Log

import com.lambdaschool.congressdataapiaccess.CongressDao

object ProfileRepository {
    fun getProfile(id: String?): LiveData<CongresspersonProfile> {
        Log.i("Repository", "Retreiving Data")
        val profileLiveData = MutableLiveData<CongresspersonProfile>()

        id?.let {
            val profile = CongresspersonProfile(CongressDao.getMemberDetails(id))
            profile.image = CongressDao.getImage(profile.id)
            profileLiveData.value = profile
        }
        return profileLiveData
    }
}
